From 036faba2adf7a9ca0e878b7fc7cafc5388edb527 Mon Sep 17 00:00:00 2001
From: Charlotte Van Petegem <charlotte.vanpetegem@ugent.be>
Date: Mon, 24 Oct 2022 09:28:30 +0200
Subject: [PATCH] zeroad: fix build

---
 pkgs/games/0ad/game.nix | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pkgs/games/0ad/game.nix b/pkgs/games/0ad/game.nix
index 488ef0c91e17f..7e1375a829f97 100644
--- a/pkgs/games/0ad/game.nix
+++ b/pkgs/games/0ad/game.nix
@@ -43,9 +43,9 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional withEditor wxGTK;
 
   NIX_CFLAGS_COMPILE = toString [
-    "-I${xorgproto}/include/X11"
-    "-I${libX11.dev}/include/X11"
-    "-I${libXcursor.dev}/include/X11"
+    "-I${xorgproto}/include"
+    "-I${libX11.dev}/include"
+    "-I${libXcursor.dev}/include"
     "-I${SDL2}/include/SDL2"
     "-I${fmt.dev}/include"
     "-I${nvidia-texture-tools.dev}/include"
