diff --git a/pkgs/tools/networking/udp2raw/default.nix b/pkgs/tools/networking/udp2raw/default.nix
index 7bbba969ef1a4..b6466ce0be57b 100644
--- a/pkgs/tools/networking/udp2raw/default.nix
+++ b/pkgs/tools/networking/udp2raw/default.nix
@@ -29,6 +29,8 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     echo 'const char *gitversion = "${version}";' > git_version.h
+    # Adress sanitization crashes the application, reported upstream at https://github.com/wangyu-/udp2raw/issues/474
+    substituteInPlace CMakeLists.txt --replace "sanitize=address," "sanitize="
   '';
 
   nativeBuildInputs = [
