diff --git a/pkgs/applications/networking/p2p/transmission-remote-gtk/default.nix b/pkgs/applications/networking/p2p/transmission-remote-gtk/default.nix
index b85a440b0ed401..77124120d5d343 100644
--- a/pkgs/applications/networking/p2p/transmission-remote-gtk/default.nix
+++ b/pkgs/applications/networking/p2p/transmission-remote-gtk/default.nix
@@ -50,7 +50,7 @@ stdenv.mkDerivation rec {
     libmrss
     libproxy
     libsoup_3
-  ];
+  ] ++ libsoup_3.propagatedUserEnvPackages;
 
   doCheck = false; # Requires network access
 
