diff --git a/nixos/modules/services/mail/dovecot.nix b/nixos/modules/services/mail/dovecot.nix
index 8d298de6945bbc..b40798ed6d588b 100644
--- a/nixos/modules/services/mail/dovecot.nix
+++ b/nixos/modules/services/mail/dovecot.nix
@@ -112,10 +112,6 @@ let
       pop3_uidl_format = %08Xv%08Xu
 
       auth_mechanisms = plain login
-
-      service auth {
-        user = root
-      }
     ''
 
     (
