diff --git a/pkgs/development/python-modules/xmlschema/default.nix b/pkgs/development/python-modules/xmlschema/default.nix
index 927296f072f7962..efb003146714648 100644
--- a/pkgs/development/python-modules/xmlschema/default.nix
+++ b/pkgs/development/python-modules/xmlschema/default.nix
@@ -35,10 +35,9 @@ buildPythonPackage rec {
   ];
 
   disabledTests = [
-    # -file://///filer01/MY_HOME/dev/XMLSCHEMA/test.xsd
-    # +file:////filer01/MY_HOME/dev/XMLSCHEMA/test.xsd
-    "test_normalize_url_slashes"
-    "test_normalize_url_with_base_unc_path"
+    # Incorrect error message in pickling test for Python 3.12 in Debian
+    # https://github.com/sissaschool/xmlschema/issues/412
+    "test_pickling_subclassed_schema__issue_263"
   ];
 
   pythonImportsCheck = [ "xmlschema" ];
