diff --git a/pkgs/by-name/ze/zeroc-ice/package.nix b/pkgs/by-name/ze/zeroc-ice/package.nix
index 9331ac1502a6f..8a6c5644381eb 100644
--- a/pkgs/by-name/ze/zeroc-ice/package.nix
+++ b/pkgs/by-name/ze/zeroc-ice/package.nix
@@ -2,10 +2,12 @@
   stdenv,
   lib,
   fetchFromGitHub,
+  fetchpatch2,
   bzip2,
   expat,
   libedit,
   lmdb,
+  mcpp,
   openssl,
   libxcrypt,
   python3, # for tests only
@@ -13,21 +15,35 @@
 }:
 
 let
-  zeroc_mcpp = stdenv.mkDerivation rec {
-    pname = "zeroc-mcpp";
-    version = "2.7.2.14";
+  mcpp' = mcpp.overrideAttrs (prevAttrs: rec {
+    pname = "mcpp-zeroc-ice";
+    version = "2.7.3";
 
     src = fetchFromGitHub {
       owner = "zeroc-ice";
       repo = "mcpp";
       rev = "v${version}";
-      sha256 = "1psryc2ql1cp91xd3f8jz84mdaqvwzkdq2pr96nwn03ds4cd88wh";
+      hash = "sha256-hZGU5mqMRTTHV2bR9uzM6ALj1sypjPxO5Ajg8aKzLxc=";
     };
 
-    configureFlags = [ "--enable-mcpplib" ];
-    installFlags = [ "PREFIX=$(out)" ];
-  };
+    # zeroc-ice's fork diverges quite a bit from upstream mcpp, so prevAttrs.patches is not used here
+    patches = [
+      # See https://github.com/zeroc-ice/mcpp/pull/12
+      (fetchpatch2 {
+        name = "mcpp-zeroc-ice-mb_init.patch";
+        url = "https://github.com/zeroc-ice/mcpp/commit/faa694a7171a06f83034fd869adc4cffa2ae0c18.patch?full_index=1";
+        hash = "sha256-TI3p/0thyvI+IgR4/rZzugzvD42Gxbeaemfa2Z5WBs4=";
+      })
+      # See https://github.com/zeroc-ice/mcpp/pull/12
+      (fetchpatch2 {
+        name = "mcpp-zeroc-ice-reserved_keywords.patch";
+        url = "https://github.com/zeroc-ice/mcpp/commit/bd5ddbde9e4ed24101cdc86007f26e95f38dd5b1.patch?full_index=1";
+        hash = "sha256-4F/hs+m0oz8ZkhDaT1b4gWC4w2bsaltShMDt2v88qt0=";
+      })
+    ];
 
+    installFlags = prevAttrs.installFlags or [ ] ++ [ "PREFIX=$(out)" ];
+  });
 in
 stdenv.mkDerivation rec {
   pname = "zeroc-ice";
@@ -41,7 +57,7 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [
-    zeroc_mcpp
+    mcpp'
     bzip2
     expat
     libedit
