diff --git a/pkgs/by-name/bl/blackshades/package.nix b/pkgs/by-name/bl/blackshades/package.nix
index b26ecf1b0b17e..7b38f01cfbc50 100644
--- a/pkgs/by-name/bl/blackshades/package.nix
+++ b/pkgs/by-name/bl/blackshades/package.nix
@@ -23,7 +23,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-W6ltmWCw7jfiTiNlh60YVF7mz//8s+bgu4F9gy5cDgw=";
   };
 
-  postUnpack = ''
+  postConfigure = ''
     ln -s ${
       zig_0_14.fetchDeps {
         inherit (finalAttrs)
diff --git a/pkgs/by-name/bo/bold/package.nix b/pkgs/by-name/bo/bold/package.nix
index b6892bc699a0d..405fd0972745d 100644
--- a/pkgs/by-name/bo/bold/package.nix
+++ b/pkgs/by-name/bo/bold/package.nix
@@ -31,7 +31,7 @@ stdenv.mkDerivation (finalAttrs: {
     })
   ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/bo/bork/package.nix b/pkgs/by-name/bo/bork/package.nix
index c2de996fe2afe..76f48aa2607ed 100644
--- a/pkgs/by-name/bo/bork/package.nix
+++ b/pkgs/by-name/bo/bork/package.nix
@@ -31,7 +31,7 @@ stdenvNoCC.mkDerivation {
 
   zigBuildFlags = [ "--release=fast" ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/fa/fancy-cat/package.nix b/pkgs/by-name/fa/fancy-cat/package.nix
index 09e69e28712d1..f9edf039d3e9c 100644
--- a/pkgs/by-name/fa/fancy-cat/package.nix
+++ b/pkgs/by-name/fa/fancy-cat/package.nix
@@ -45,7 +45,7 @@ stdenv.mkDerivation (finalAttrs: {
     libz
   ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./build.zig.zon.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/fl/flow-control/package.nix b/pkgs/by-name/fl/flow-control/package.nix
index 159ad3f3a4ca2..a6030765597dc 100644
--- a/pkgs/by-name/fl/flow-control/package.nix
+++ b/pkgs/by-name/fl/flow-control/package.nix
@@ -16,7 +16,7 @@ stdenv.mkDerivation (finalAttrs: {
     tag = "v${finalAttrs.version}";
     hash = "sha256-868FK3wr/fjXzrQJ4YVDBvzNuX818lufEx/K0fvJdWo=";
   };
-  postPatch = ''
+  postConfigure = ''
     ln -s ${
       callPackage ./build.zig.zon.nix {
         zig = zig_0_15;
diff --git a/pkgs/by-name/he/hevi/package.nix b/pkgs/by-name/he/hevi/package.nix
index 91c72dda05839..b1a6bae5167bb 100644
--- a/pkgs/by-name/he/hevi/package.nix
+++ b/pkgs/by-name/he/hevi/package.nix
@@ -24,7 +24,7 @@ stdenv.mkDerivation (finalAttrs: {
     zig
   ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/li/linuxwave/package.nix b/pkgs/by-name/li/linuxwave/package.nix
index 328365d834bb3..973924615c961 100644
--- a/pkgs/by-name/li/linuxwave/package.nix
+++ b/pkgs/by-name/li/linuxwave/package.nix
@@ -22,7 +22,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-OuD5U/T3GuCQrzdhx01NXPSXD7pUAvLnNsznttJogz8=";
   };
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/ls/lsr/package.nix b/pkgs/by-name/ls/lsr/package.nix
index 01c70c41c6965..a2f56978f9568 100644
--- a/pkgs/by-name/ls/lsr/package.nix
+++ b/pkgs/by-name/ls/lsr/package.nix
@@ -25,7 +25,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-VeB0R/6h9FXSzBfx0IgpGlBz16zQScDSiU7ZvTD/Cds=";
   };
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/ly/ly/package.nix b/pkgs/by-name/ly/ly/package.nix
index 7b937ad12d5ee..3d25d1a48cb3e 100644
--- a/pkgs/by-name/ly/ly/package.nix
+++ b/pkgs/by-name/ly/ly/package.nix
@@ -36,7 +36,7 @@ stdenv.mkDerivation (finalAttrs: {
   ]
   ++ (lib.optionals x11Support [ libxcb ]);
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${
       callPackage ./deps.nix {
         inherit zig;
diff --git a/pkgs/by-name/od/odiff/package.nix b/pkgs/by-name/od/odiff/package.nix
index 9341e8a6fd193..4aca5a474e5fb 100644
--- a/pkgs/by-name/od/odiff/package.nix
+++ b/pkgs/by-name/od/odiff/package.nix
@@ -23,7 +23,7 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-gCF+CInczBJfDyZgxEQor5C/OSxKciCu9gbZanaE/nA=";
   };
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./build.zig.zon.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/ri/river-classic/package.nix b/pkgs/by-name/ri/river-classic/package.nix
index 7385f480bd471..802fa35f1da52 100644
--- a/pkgs/by-name/ri/river-classic/package.nix
+++ b/pkgs/by-name/ri/river-classic/package.nix
@@ -59,8 +59,6 @@ stdenv.mkDerivation (finalAttrs: {
   ]
   ++ lib.optional xwaylandSupport libx11;
 
-  dontConfigure = true;
-
   zigBuildFlags = [
     "--system"
     "${finalAttrs.deps}"
diff --git a/pkgs/by-name/ri/rivercarro/package.nix b/pkgs/by-name/ri/rivercarro/package.nix
index c88e5ec615898..775b4c510430c 100644
--- a/pkgs/by-name/ri/rivercarro/package.nix
+++ b/pkgs/by-name/ri/rivercarro/package.nix
@@ -34,7 +34,7 @@ stdenv.mkDerivation (finalAttrs: {
     zig
   ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/by-name/su/superhtml/package.nix b/pkgs/by-name/su/superhtml/package.nix
index 2ade74c68c094..e903bede55d6d 100644
--- a/pkgs/by-name/su/superhtml/package.nix
+++ b/pkgs/by-name/su/superhtml/package.nix
@@ -23,7 +23,7 @@ stdenv.mkDerivation (finalAttrs: {
     zig
   ];
 
-  postPatch = ''
+  postConfigure = ''
     ln -s ${callPackage ./deps.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
   '';
 
diff --git a/pkgs/development/tools/zls/default.nix b/pkgs/development/tools/zls/default.nix
index 849309893cdf6..9e6e2b4eb207e 100644
--- a/pkgs/development/tools/zls/default.nix
+++ b/pkgs/development/tools/zls/default.nix
@@ -40,7 +40,7 @@ lib.mapAttrs (_: extension: stdenv.mkDerivation (lib.extends common extension))
 
     nativeBuildInputs = [ zig_0_14 ];
 
-    postPatch = ''
+    postConfigure = ''
       ln -s ${callPackage ./deps_0_14.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
     '';
   };
@@ -58,7 +58,7 @@ lib.mapAttrs (_: extension: stdenv.mkDerivation (lib.extends common extension))
 
     nativeBuildInputs = [ zig_0_15 ];
 
-    postPatch = ''
+    postConfigure = ''
       ln -s ${callPackage ./deps_0_15.nix { }} $ZIG_GLOBAL_CACHE_DIR/p
     '';
   };
