diff --git a/nixos/modules/services/misc/forgejo.nix b/nixos/modules/services/misc/forgejo.nix
index 5ed68d6df6199..ca6cb0a4feaab 100644
--- a/nixos/modules/services/misc/forgejo.nix
+++ b/nixos/modules/services/misc/forgejo.nix
@@ -801,9 +801,10 @@ in
       // lib.listToAttrs (map (e: lib.nameValuePair e.env "%d/${e.env}") secrets);
     };
 
-    services.openssh.settings.AcceptEnv = mkIf (!cfg.settings.server.START_SSH_SERVER or false) [
-      "GIT_PROTOCOL"
-    ];
+    services.openssh.extraConfig = mkIf (!cfg.settings.server.START_SSH_SERVER or false) ''
+      Match User ${cfg.user}
+        AcceptEnv GIT_PROTOCOL
+    '';
 
     users.users = mkIf (cfg.user == "forgejo") {
       forgejo = {
