diff --git a/pkgs/data/themes/arc/default.nix b/pkgs/data/themes/arc/default.nix
index fff5e4bf41a..0ece2d6e0e0 100644
--- a/pkgs/data/themes/arc/default.nix
+++ b/pkgs/data/themes/arc/default.nix
@@ -1,7 +1,8 @@
 { lib, stdenv
 , fetchFromGitHub
 , sassc
-, autoreconfHook
+, meson
+, ninja
 , pkg-config
 , gtk3
 , gnome
@@ -13,7 +14,7 @@
 
 stdenv.mkDerivation rec {
   pname = "arc-theme";
-  version = "20210127";
+  version = "20210412";
 
   src = fetchFromGitHub {
     owner = "jnsh";
@@ -23,12 +24,15 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [
-    autoreconfHook
+    meson
+    ninja
     pkg-config
     sassc
     optipng
     inkscape
     gtk3
+    gnome.gnome-shell
+    cinnamon.cinnamon-common
   ];
 
   propagatedUserEnvPkgs = [
@@ -43,14 +47,18 @@ stdenv.mkDerivation rec {
     export HOME="$NIX_BUILD_ROOT"
   '';
 
-  configureFlags = [
-    "--with-cinnamon=${cinnamon.cinnamon-common.version}"
-    "--with-gnome-shell=${gnome.gnome-shell.version}"
-    "--disable-unity"
+  mesonFlags = [
+    "-Dthemes=cinnamon,gnome-shell,gtk2,gtk3,plank,xfwm"
+    "-Dvariants=light,darker,dark,lighter"
+    "-Dtransparency=true"
+    "-Dcinnamon_version=${cinnamon.cinnamon-common.version}"
+    "-Dgnome_shell_version=${lib.elemAt (lib.splitString "-" gnome.gnome-shell.version) 0}"
+    "-Dgtk3_version=${gtk3.version}"
+    "-Dgnome_shell_resource=false"
   ];
 
   postInstall = ''
-    install -Dm644 -t $out/share/doc/${pname} AUTHORS *.md
+    install -Dm644 -t $out/share/doc/${pname} $src/AUTHORS $src/*.md
   '';
 
   meta = with lib; {
