diff --git a/pkgs/servers/mastodon/default.nix b/pkgs/servers/mastodon/default.nix
index 8f34d284df2..9d1a25e4e97 100644
--- a/pkgs/servers/mastodon/default.nix
+++ b/pkgs/servers/mastodon/default.nix
@@ -31,9 +31,10 @@ stdenv.mkDerivation rec {
         cp -a "$gem/" "$gem.new"
         rm "$gem"
         # needed on macOS, otherwise the mv yields permission denied
-        chmod +w "$gem.new"
+        chmod -R +w "$gem.new"
         mv "$gem.new" "$gem"
       done
+      cp "$out"/lib/ruby/gems/*/gems/blurhash-*/lib/encode.so "$out"/lib/ruby/gems/*/gems/blurhash-*/ext/blurhash/
     '';
   };
 
